src/index/IndexImpl.cpp

```diff
@@ -950,6 +943,9 @@ void IndexImpl::createFromOnDiskIndex(const std::string& onDiskBase,
     permutation.loadFromDisk(onDiskBase_, isInternalId,
                              loadInternalPermutation);
     setMetadata(permutation);
+    if (loadInternalPermutation) {
+      setMetadata(permutation.internalPermutation());
+    }
```

Comment:
This can be done directly by the permutation (if the permutation has an internal permutation, then it also registers that internal perm with the delta triples.